src/backend/OverlayFS.ts

```diff
@@ -2,34 +2,55 @@ import file_system = require('../core/file_system');
 import {ApiError, ErrorCode} from '../core/api_error';
 import {FileFlag, ActionType} from '../core/file_flag';
 import util = require('../core/util');
-import file = require('../core/file');
-import Stats from '../core/node_fs_stats';
-import preload_file = require('../generic/preload_file');
+import {File} from '../core/file';
+import {default as Stats, FileType} from '../core/node_fs_stats';
+import {PreloadFile} from '../generic/preload_file';
+import {LockedFS} from '../generic/locked_fs';
 import path = require('path');
 let deletionLogPath = '/.deletedFiles.log';
 
 /**
  * Given a read-only mode, makes it writable.
  */
 function makeModeWritable(mode: number): number {
-  return 0x92 | mode;
+  return 0o222 | mode;
```

Comment:
Nice; I didn't know you could do octals like this. I would love to see these replacing all of the hex values across the codebase.